From b138cfd473ab65282af07d400fc95a8a5bc7e22e Mon Sep 17 00:00:00 2001
From: Vladimir 'phcoder' Serbinenko <phcoder@gmail.com>
Date: Sun, 24 Mar 2013 14:03:33 +0100
Subject: [PATCH 231/482] 	* util/grub.d/20_linux_xen.in: Automatically
 add no-real-mode edd=off on 	non-BIOS platforms.

---
 ChangeLog                   | 5 +++++
 util/grub.d/20_linux_xen.in | 7 ++++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index 43dcf93..981991b 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,10 @@
 2013-03-24  Vladimir Serbinenko  <phcoder@gmail.com>
 
+	* util/grub.d/20_linux_xen.in: Automatically add no-real-mode edd=off on
+	non-BIOS platforms.
+
+2013-03-24  Vladimir Serbinenko  <phcoder@gmail.com>
+
 	* grub-core/Makefile.core.def (vga): Disable on coreboot and multiboot
 	platforms.
 
diff --git a/util/grub.d/20_linux_xen.in b/util/grub.d/20_linux_xen.in
index ac05ee4..6651cbc 100644
--- a/util/grub.d/20_linux_xen.in
+++ b/util/grub.d/20_linux_xen.in
@@ -121,7 +121,12 @@ linux_entry ()
   lmessage="$(gettext_printf "Loading Linux %s ..." ${version})"
   sed "s/^/$submenu_indentation/" << EOF
 	echo	'$(echo "$xmessage" | grub_quote)'
-	multiboot	${rel_xen_dirname}/${xen_basename} placeholder ${xen_args}
+        if [ "\$grub_platform" = "pc" -o "\$grub_platform" = "" ]; then
+            xen_rm_opts=
+        else
+            xen_rm_opts="no-real-mode edd=off"
+        fi
+	multiboot	${rel_xen_dirname}/${xen_basename} placeholder ${xen_args} \${xen_rm_opts}
 	echo	'$(echo "$lmessage" | grub_quote)'
 	module	${rel_dirname}/${basename} placeholder root=${linux_root_device_thisversion} ro ${args}
 EOF
-- 
1.8.2.1

